Fix retrieval of login.saml.providers

strehle opened the pull request:
The map can be incomplete if a dot is in key

e.g.
https://stackoverflow.com/questions/44014910/preventing-spring-boot-from-creating-nested-map-from-dot-separated-key-in-applic

strehle committed:
Fix retrieval of login.saml.providers

The map can be incomplete if a dot is in key

e.g.
https://stackoverflow.com/questions/44014910/preventing-spring-boot-from-creating-nested-map-from-dot-separated-key-in-applic

strehle committed:
add test

strehle committed:
test coverage

strehle committed:
cleanup fix.

strehle committed:
Merge remote-tracking branch 'origin/develop' into fix/saml-provider-alias-with-dot

strehle committed:
sonar fix for test

strehle committed:
Use another name for provider Map

duanemay commented on server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SamlConfigProps.java:
```diff
@@ -106,18 +116,21 @@ public class SamlConfigProps {
 
     /**
      * Deprecated but sill working: login.serviceProviderKey
+     * WILL be REMOVED with UAA 78.x
```
Should these be annotated @Deprecated(since = "77.20.0", forRemoval = true)

duanemay commented on server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SamlConfigProps.java:
```diff
@@ -130,4 +143,23 @@ public class SamlConfigProps {
         return keys != null ? keys.get(activeKeyId) : null;
     }
+
+    public Map<String, Map<String, Object>> getEnvironmentProviders() {
```
Class is marked with @Data, so this is auto generated, and can be removed

duanemay commented on server/src/test/java/org/cloudfoundry/identity/uaa/provider/saml/SamlConfigurationTest.java:
```diff
@@ -0,0 +1,73 @@
+
+    @Test
+    void bindFromEnvironment() {
+        //@formatter:off
```
At least with Idea, text blocks do not get reformatted

strehle committed:
Remove custom value annotations

Move the custom part into setEnvironment since we have here
other customer settings

strehle committed:
Merge remote-tracking branch 'origin/develop' into fix/saml-provider-alias-with-dot

strehle committed:
Test fix

strehle committed:
sonar cleanup

strehle commented:
Sonar clean, now ready for review

duanemay committed:
Minor Updates
